### PR TITLE
Feature/pla 1524 docs faq

### PR DESCRIPTION
**Pull request by ventura-rivera**
# Citrine Python PR

## Description 
New FAQ documentation that specifies cyclic data patterns are prohibited. This addresses issue [PLA-1524](https://citrine.atlassian.net/browse/PLA-1524)

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [x] I have communicated the downstream consequences of the PR to others.

### CODE REVIEW

File: docs/source/FAQ/prohibited_data_patterns.rst [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+>>> procsp = dataset.process_specs.register(ProcessSpec(name='test proc'))
+>>> matsp = dataset.material_specs.register(MaterialSpec(name='test mat',process=procsp))
+>>> ingsp = dataset.ingredient_specs.register(IngredientSpec(name='ing',material=matsp, process=procsp))
+>>>
+
+In this example, the following steps occurred:
+
+1. A process was created
+2. A material was assigned the above process. (which makes it an output material to that process)
```
Comment (maxhutch) [inline docs/source/FAQ/prohibited_data_patterns.rst]:
```suggestion
2. A material was assigned the above process, which makes it an output material to that process.
```

File: docs/source/FAQ/prohibited_data_patterns.rst [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+Data objects should **NOT** be linked in a cyclic pattern. Consider the following example (where the citrine session is created and dataset already defined):
+
+>>> procsp = dataset.process_specs.register(ProcessSpec(name='test proc'))
+>>> matsp = dataset.material_specs.register(MaterialSpec(name='test mat',process=procsp))
+>>> ingsp = dataset.ingredient_specs.register(IngredientSpec(name='ing',material=matsp, process=procsp))
+>>>
+
+In this example, the following steps occurred:
+
+1. A process was created
```
Comment (maxhutch) [inline docs/source/FAQ/prohibited_data_patterns.rst]:
```suggestion
1. A process was created.
```

File: docs/source/FAQ/prohibited_data_patterns.rst [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+2. A material was assigned the above process. (which makes it an output material to that process)
+3. An ingredient was assigned the above material and process. (which makes it an input to that process belonging to that material)
```
Comment (maxhutch) [inline docs/source/FAQ/prohibited_data_patterns.rst]:
```suggestion
3. An ingredient was assigned the above material and process, which makes it an input to that process belonging to that material.
```

File: docs/source/FAQ/prohibited_data_patterns.rst [hunk 4]
```diff
@@ -0,0 +1,38 @@
+========================
+Prohibited Data Patterns
+========================
+
+Summary
+=======
+
+This is a disclaimer on our end to let you know that there are some patterns that may corrupt your data and eliminate
```
Comment (maxhutch) [inline docs/source/FAQ/prohibited_data_patterns.rst]:
```suggestion
This is a disclaimer on our end to let you know that there are some invalid usage patterns that are not prevented by the API but may corrupt your data and eliminate 
```

File: docs/source/FAQ/prohibited_data_patterns.rst [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+>>>
+
+In this example, the following steps occurred:
+
+1. A process was created
+2. A material was assigned the above process. (which makes it an output material to that process)
+3. An ingredient was assigned the above material and process. (which makes it an input to that process belonging to that material)
+
+This is a cyclic pattern because the material is an output to the process, but the ingredient (from the same material) is an input to the same process. 
+In general, an object should not reference any other object in it's history to make a cyclic pattern.
```
Comment (maxhutch) [inline docs/source/FAQ/prohibited_data_patterns.rst]:
```suggestion
In general, an object should not reference any other object in it's history to make a cyclic pattern.
It does not make sense for a material to be used as an ingredient in its own history.
In other words, the data model does not support time traveling materials.
```

**Comment by maxhutch** [comment]
Note: the test failures are due to a change in the `taurus:develop` that's addressed here:
https://github.com/CitrineInformatics/citrine-python/pull/70

File: docs/source/FAQ/prohibited_data_patterns.rst (hunk 5 above)
Comment (ventura-rivera) [reply docs/source/FAQ/prohibited_data_patterns.rst]: hahaha this last line is a great cherry on top 😂

**Pull request merged**